src/clj/game/core/actions.clj

```diff
@@ -37,6 +37,19 @@
           click-states (vec (take-last 4 (conj (:click-states @state) state')))]
       (swap! state assoc :click-states click-states))))
 
+(defn- valid-prompt-state?
+  [state side]
+  (let [prompt-type (get-in @state [side :prompt-state :prompt-type])]
+    (or (= nil prompt-type)
+        (= :run prompt-type)
+        (= :prevent prompt-type))))
+
+(defn- no-prompt?
```

Comment:
This implementation does not check  that there's no prompt, so the name should change or the impl should change.